api/app/reviews/api.py

```diff
@@ -1,7 +1,134 @@
+    'order': fields.Integer,
+    'validation_regex': fields.String,
+    'validation_text': fields.String,
+    'weight': fields.Float
+}
+
+review_form_fields = {
+    'id': fields.Integer,
+    'application_form_id': fields.Integer,
+    'is_open': fields.Boolean,
+    'deadline': fields.DateTime('iso8601'),
+    'review_questions': fields.List(fields.Nested(review_question_fields))
+}
+
+answer_fields = {
+    'id': fields.Integer,
+    'question_id': fields.Integer,
+    'question': fields.String(attribute='question.headline'),
+    'value': fields.String
+}
+
+response_fields = {
+    'id': fields.Integer,
+    'application_form_id': fields.Integer,
+    'user_id': fields.Integer,
+    'is_submitted': fields.Boolean,
+    'submitted_timestamp': fields.DateTime(dt_format='iso8601'),
+    'is_withdrawn': fields.Boolean,
+    'withdrawn_timestamp': fields.DateTime(dt_format='iso8601'),
+    'started_timestamp': fields.DateTime(dt_format='iso8601'),
+    'answers': fields.List(fields.Nested(answer_fields))
+}
+
+user_fields = {
+    'nationality_country': fields.String(attribute='nationality_country.name'),
+    'residence_country': fields.String(attribute='residence_country.name'),
+    'affiliation': fields.String,
+    'department': fields.String,
+    'user_category': fields.String(attribute='user_category.name')
+}
+
+review_response_fields = {
+    'review_form': fields.Nested(review_form_fields),
+    'response': fields.Nested(response_fields),
+    'user': fields.Nested(user_fields),
+    'reviews_remaining_count': fields.Integer
+}
+
+class ReviewResponseUser():
+    def __init__(self, review_form, response, reviews_remaining_count):
+        self.review_form = review_form
+        self.response = response
+        self.user = None if response is None else response.user
+        self.reviews_remaining_count = reviews_remaining_count
+
+class ReviewAPI(ReviewMixin, restful.Resource):
+
+    @auth_required
+    @marshal_with(review_response_fields)
     def get(self):
-        return 200
\ No newline at end of file
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+        
+        review_form = db.session.query(ReviewForm)\
+                        .join(ApplicationForm, ApplicationForm.id==ReviewForm.application_form_id)\
+                        .filter_by(event_id=event_id)\
+                        .first()
+        if review_form is None:
+            return EVENT_NOT_FOUND
+
+        reviews_remaining_count = db.session.query(func.count(ResponseReviewer.id))\
+                        .filter_by(reviewer_user_id=g.current_user['id'])\
+                        .join(Response)\
+                        .filter_by(is_withdrawn=False, application_form_id=review_form.application_form_id, is_submitted=True)\
+                        .outerjoin(ReviewResponse, and_(ReviewResponse.response_id==ResponseReviewer.response_id, ReviewResponse.reviewer_user_id==g.current_user['id']))\

```

Comment:
Would we need to do this:
```
ReviewResponse.reviewer_user_id==g.current_user['id'])
```

If we have done,
```
.filter_by(reviewer_user_id=g.current_user['id'])
``` 
above?

Reply:
So both are actually required.

The `.filter_by` is to limit the ResponseReviewers to just the logged in user. The reason is that we're doing an outer join later which means all the ResponseReviewers are preserved and so every single one of those are counted (instead of just the logged in user's).

The `ReviewResponse.reviewer_user_id==g.current_user['id'])` in the outer join is to make sure that if two reviewers are assigned to a response and one of them completes a review, the query will still bring back the response for the second reviewer. (I'm struggling to explain this better so let me know if it doesn't make sense. This one was a bit confusing for me to figure out.)

Reply:
No it makes sense thanks.